shared/src/main/scala-3/org/scalamock/clazz/MockMaker.scala

```diff
@@ -124,7 +124,8 @@ private[clazz] object MockMaker:
               privateWithin = Symbol.noSymbol
             )
 
-        List(mockFunctionSym, overrideSym)
+        if (definition.symbol.flags.is(Flags.Artifact)) Nil
```

Comment:
Can we move this one [here](https://github.com/paulbutcher/ScalaMock/blob/master/shared/src/main/scala-3/org/scalamock/clazz/Utils.scala#L177)?

Reply:
I would consider moving it there if we can

Reply:
Suspect that should be fine, will try that